nextcord/errors.py

```diff
@@ -78,6 +80,10 @@ def _flatten_error_dict(d: Dict[str, Any], key: str = "") -> Dict[str, str]:
     return dict(items)
 
 
+class HTTPCancelled(DiscordException):
+    """Exception that's raised when an HTTP request is internally cancelled."""
```

Comment:
Would just love maybe a `#` comment explaining if an end user should ever see this or not? More so my curiosity as to expectations here then anything

Reply:
As it stands right now, the end user can see it. Currently it's raised when someone does a request with `retry_request=False` and the ratelimit is already capped out, but I can make a more specific error that subclasses `HTTPCancelled` if that's desired.

Reply:
My thinking here is just that as an end user, seeing an error with "internally cancelled" doesn't really tell me if I need to handle the error, or if its fine to ignore if that makes sense?

Maybe just another line saying essentially what you said would be nice. Something like: "... This exception indicates that a request was made to a route which exceeded the ratelimited. As a result, the request was cancelled."

Reply:
Made some changes and added a new exception